build.rs

```diff
@@ -13,4 +13,19 @@ fn main() {
     {
         println!("cargo::rustc-cfg=feature=\"include-ui\"");
     }
+    // Create a new, empty icon collection
+    let mut icon_dir = ico::IconDir::new(ico::ResourceType::Icon);
+    // Read a PNG file from disk and add it to the collection
+    let file = std::fs::File::open("assets/icon.png").expect("Could not find assets/icon.png");
+    let image = ico::IconImage::read_png(file).expect("Could not read PNG file");
+    icon_dir.add_entry(
+        ico::IconDirEntry::encode(&image).expect("Could not add PNG file to icon collection"),
+    );
+    // Write the ICO file to disk
+    let file = std::fs::File::create("assets/icon.ico").expect("Could not create assets/icon.ico");
+    icon_dir
+        .write(file)
+        .expect("Could not write assets/icon.ico to disk.");
```

Comment:
Since the .ico file hangs around after build, you'll have to add it to the gitignore.